Source/WebCore/editing/CompositeEditCommand.cpp

```diff
@@ -802,6 +802,8 @@ Position CompositeEditCommand::replaceSelectedTextInNode(const String& text)
         return Position();
 
     RefPtr<Text> textNode = start.containerText();
+    if (start.offsetInContainerNode() >= static_cast<int>(textNode->length()) || end.offsetInContainerNode() > static_cast<int>(textNode->length()))
```

Comment:
This looks roughly correct. Not sure we have enough test coverage, though.

Does this function ever get used to replace a zero-length selection, like putting text at an insertion point? If so, then we should allow the start offset to be equal to the length, and it should be > rather than >=.

There seem to be some missing checks here. For example, no check that `end.offsetInContainerNode() >= start.offsetInContainerNode()`. But maybe I’m misunderstanding what the foundation is for needing each check.